tests/TestCase/Validation/ValidatorTest.php

```diff
@@ -164,6 +189,33 @@ public function testErrorsWithPresenceRequired()
     }
 
     /**
+     * Test that errors() can work with nested data.
+     *
+     * @return void
+     */
+    public function testErrorsWithNestedFields()
+    {
+        $validator = new Validator;
+        $validator->add('user.username', 'letter', ['rule' => 'alphanumeric']);
+        $validator->add('comments.0.comment', 'letter', ['rule' => 'alphanumeric']);
```

Comment:
It doesn't have to be too invasive. In my mind a nested validator was just a custom rule from a custom provider.